ui/app/src/components/UsersGrid/UserInfoDialog.tsx

```diff
@@ -0,0 +1,278 @@
+import DialogBody from '../Dialogs/DialogBody';
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+import { Divider } from '@material-ui/core';
+import User from '../../models/User';
+import Typography from '@material-ui/core/Typography';
+import Avatar from '@material-ui/core/Avatar';
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+import IconButton from '@material-ui/core/IconButton';
+import DeleteRoundedIcon from '@material-ui/icons/DeleteRounded';
+import EditRoundedIcon from '@material-ui/icons/EditRounded';
+import CloseRoundedIcon from '@material-ui/icons/CloseRounded';
+import Input from '@material-ui/core/Input';
+import { useSpreadState } from '../../utils/hooks';
+import { update } from '../../services/users';
+import { useDispatch } from 'react-redux';
+import { showErrorDialog } from '../../state/reducers/dialogs/error';
+import { showSystemNotification } from '../../state/actions/system';
+import SecondaryButton from '../SecondaryButton';
+import PrimaryButton from '../PrimaryButton';
+import clsx from 'clsx';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+const styles = makeStyles(() =>
+  createStyles({
+    header: {
+      padding: '30px 40px',
+      display: 'flex',
+      alignItems: 'center'
+    },
+    avatar: {
+      marginRight: '30px',
+      width: '90px',
+      height: '90px'
+    },
+    actions: {
+      marginLeft: 'auto'
+    },
+    userInfo: {},
+    body: {
+      padding: 0
+    },
+    section: {
+      padding: '30px 40px',
+      '&.noPaddingBottom': {
+        paddingBottom: 0
+      }
+    },
+    row: {
+      display: 'flex',
+      padding: '15px 0',
+      alignItems: 'center'
+    },
+    formActions: {
+      display: 'flex',
+      paddingBottom: '20px',
+      '&.hidden': {
+        visibility: 'hidden'
+      },
+      '& button:first-child': {
+        marginLeft: 'auto',
+        marginRight: '10px'
+      }
+    },
+    label: {
+      flexBasis: '180px'
+    },
+    sectionTitle: {
+      textTransform: 'uppercase',
+      marginBottom: '10px'
+    },
+    inputRoot: {
+      pointerEvents: 'none'
+    },
+    readOnlyInput: {
+      background: 'none',
+      borderColor: 'transparent',
+      '&:focus': {
+        boxShadow: 'none'
+      }
+    }
+  })
+);
+
+const translations = defineMessages({
+  userUpdated: {
+    id: 'userInfoDialog.userUpdated',
+    defaultMessage: 'User updated successfully'
+  }
+});
+
+interface UserInfoDialogProps {
+  open: boolean;
+  onClose(): void;
+  onUserEdited(): void;
+  user: User;
+}
+
+export function UserInfoDialog(props: UserInfoDialogProps) {
+  const { open, onClose, onUserEdited } = props;
+  const [editMode, setEditMode] = useState(false);
+  const [user, setUser] = useSpreadState(null);
+  const [lastSavedUser, setLastSavedUser] = useState(null);
+  const [inProgress, setInProgress] = useState(false);
+  const [dirty, setDirty] = useState(false);
```

Comment:
yes because we need to keep the user being edited, and the original user, to do a rollback with the cancel button,
also the dirty helps to enable or disable the save button